doajtest/helpers.py

```diff
@@ -199,6 +200,16 @@ def _make_and_push_test_context(self, path="/", acc=None):
 
         return ctx
 
+    @staticmethod
+    def fix_es_mapping():
+        """
+        you need to call this method if you get some errors like:
+        ESMappingMissingError - 'reason': 'No mapping found for [field]
+
+        :return:
+        """
+        models.Article(**ArticleFixtureFactory.make_article_source()).save(blocking=True)
```

Comment:
If the mappings aren't properly initialised we have a problem elsewhere, I would have thought.  It seems you are only seeing this on Articles, what is the issue that caused you to need to write this bit of code?

Reply:
you can see the exception by adding following test cases in any unit test file 
```
    def test_fail(self):
        # from doajtest.fixtures import ArticleFixtureFactory
        # from portality.models import Article
        # models.Article(**ArticleFixtureFactory.make_article_source()).save(blocking=True)
        q = {'query': {'bool': {'must': [{'terms': {'admin.application_status.exact': ['revisions_required', 'pending', 'in progress', 'completed', 'on hold', 'ready', 'draft']}}], 'filter': [{'term': {'admin.owner.exact': 'publisher'}}, {'term': {'admin.application_type.exact': 'new_application'}}]}}, 
             'size': 50, 'sort': [{'last_updated': {'order': 'desc'}}], '_source': {}, 'track_total_hits': True}
        for a in Article.query(q=q):
            print(a)
```


I think `initialise_index` is not prefect and the reason why no one found out this problem is because no one will drop all index in your production environment. 
every time you drop all index in your local, you will run import data and I think the import data fix the mapping.

Reply:
Ok, I see the problem.  All of the types should push a mapping, but I guess the error here is that the query is being done on an empty index with none of the queried fields already populated?

I've seen this before, and there isn't a great fix for it, that's why we started to move the objects over to `seamless` which supports the generation of the full mapping, but we didn't do that yet for Articles.

Your fix is the best alternative that I've used, though you may want to push and then delete the article straight after so that it doesn't confuse any down-stream tests.  You could also do this in `initialise_index` so that it's true of all empty systems.

Reply:
### `solution A`
```
models.Article(**ArticleFixtureFactory.make_article_source()).save(blocking=True)
```
is temporary solution for testing environment, 
I don't know too many about Elasticsearch,  not sure is that good for `initialise_index` and is it fixing the root cause.

let's study this issues in new branch, if we can't found other better solution than `solution A` will be used.

Reply:
ok, agreed, we shouldn't do this in `initialise_index` now, and we should actually just move Articles over to use seamless, then this problem will be properly dealt with.

Only thing I'd suggest now then, is for you to do

```python
article = models.Article(**ArticleFixtureFactory.make_article_source())
article.save(blocking=True)
article.delete()
```

Then you have an empty index with the mappings initialised